### PR TITLE
feat(instructions): add endpoint to fetch current instruction status

### PR DESCRIPTION
**Pull request by VictorVicente**

### CODE REVIEW

File: src/api/entities/Instruction/index.ts [hunk 1]
```diff
@@ -307,6 +321,42 @@ export class Instruction extends Entity<UniqueIdentifiers> {
...
+    if (eventIdentifier) {
+      return {
+        status: InstructionStatus.Executed,
+        eventIdentifier,
+      };
+    }
+
+    throw new PolymeshError({
+      code: ErrorCode.FatalError,
+      message: "It wasn't possible to determine current status for this instruction",
```
Comment (monitz87) [inline src/api/entities/Instruction/index.ts:356]:
```suggestion
      message: "It isn't possible to determine the current status of this Instruction",
```

File: src/api/entities/Instruction/types.ts [hunk 2]
```diff
@@ -52,3 +54,8 @@ export interface InstructionAffirmation {
   identity: Identity;
   status: AffirmationStatus;
 }
+
+export interface InstructionStatusResult {
+  status: InstructionStatus;
+  eventIdentifier?: EventIdentifier;
+}
```
Comment (monitz87) [inline src/api/entities/Instruction/types.ts:61]: For more type safety:

```
export type InstructionStatusResult = {
  status: InstructionStatus.Pending;
} | {
  staus: Exclude<InstructionStatus, InstructionStatus.Pending>;
  eventIdentifier: EventIdentifier;
}
```

File: src/api/entities/Instruction/index.ts [hunk 3]
```diff
@@ -181,15 +195,15 @@ export class Instruction extends Entity<UniqueIdentifiers> {
...
+    if (status === InternalInstructionStatus.Unknown) {
       throw new PolymeshError({
         code: ErrorCode.FatalError,
         message: notPendingMessage,
       });
     }
 
     const details = {
-      status,
+      status: InstructionStatus.Pending,
```
Comment (monitz87) [inline src/api/entities/Instruction/index.ts:206]: Why is the explicit assignment necessary? Isn't it enough to use `status`?

File: src/api/entities/Instruction/index.ts [hunk 4]
```diff
@@ -307,6 +321,42 @@ export class Instruction extends Entity<UniqueIdentifiers> {
     };
   }
 
+  /**
+   * Retrieve current status of this Instruction
```
Comment (monitz87) [inline src/api/entities/Instruction/index.ts:325]: The middleware note is missing

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/entities/Instruction/index.ts (hunk 3 above)
Comment (VictorVicente) [reply src/api/entities/Instruction/index.ts:206]: `status` variable's type is `InternalInstructionStatus` which represents `Pending` or `Unknown`. 
`status` in details is `InstructionStatus` which represents `Pending`, `Executed` or `Failed`. 
I had to choose to force `Pending` typing or this explicit assignment which sounds like the best option for me.

**Review by monitz87 — APPROVED** [review]

**Pull request merged**